quic/s2n-quic-core/src/transport/parameters/mod.rs

```diff
@@ -1063,6 +1070,138 @@ impl From<connection::id::LocalId> for InitialSourceConnectionId {
 connection_id_parameter!(RetrySourceConnectionId, LocalId, 0x10);
 optional_transport_parameter!(RetrySourceConnectionId);
 
+/// Used by the client to indicate which versions of s2n-quic-dc it supports
+/// and by the server to indicate which version it is using
+#[derive(Clone, Copy, Debug, Default, PartialEq, PartialOrd, Eq, Ord)]
+pub struct DcSupportedVersions {
+    len: u8,
+    versions: [VarInt; DC_SUPPORTED_VERSIONS_MAX_LEN as usize],
+}
+// The maximum number of supported versions that may be transmitted using
+// the `DcSupportedVersions` transport parameter
+const DC_SUPPORTED_VERSIONS_MAX_LEN: u8 = 4;
+
+impl DcSupportedVersions {
+    /// Create a `DcSupportedVersions` for the given `supported_versions` the client supports
+    pub fn for_client<I: IntoIterator<Item = u32>>(supported_versions: I) -> Self {
+        let mut versions = [VarInt::default(); DC_SUPPORTED_VERSIONS_MAX_LEN as usize];
+        let mut len = 0;
+
+        for (index, version) in supported_versions
+            .into_iter()
+            .filter(|&version| version > 0)
```

Comment:
`0` is valid, but I don't need to waste space storing it, so I filter them out so they don't count against the actual versions. I have a test for that: https://github.com/aws/s2n-quic/pull/2193/files#diff-34ea008480e49b70923ff910908ec7757592fc48e3aa34dc7677f17f344c923fR336

Reply:
Not sure I follow... If you don't store it then it doesn't get serialized and the peer won't see it and won't be able to pick `0` as the negotiated version.

Reply:
I wasn't thinking `0` would be a valid negotiated version, since it indicated the end of the list of values before. Now that we have a `len`, `0` is a valid value